lib/jake.ex

```diff
@@ -11,46 +12,80 @@ defmodule Jake do
     "string"
   ]
 
-  def gen(%{"anyOf" => options} = spec) when is_list(options) do
+  def generator(schema) do
+    StreamData.sized(fn size ->
+      %Context{root: schema, child: schema, size: size} |> gen_lazy()
+    end)
+  end
+
+  def gen_lazy(context) do
+    StreamData.bind(
+      get_lazy_streamkey(context),
+      fn {child, size} ->
+        gen(child, %{context | child: child, size: size}) |> StreamData.resize(size)
+      end
+    )
+  end
+
+  def get_lazy_streamkey(context) do
+    {child, ref} =
+      get_in(context.child, ["$ref"]) |> Jake.Ref.expand_ref(context.child, context.root)
```

Comment:
should expand_ref take the context and return a new context with child expanded?

Reply:
It is possible to pass single context parameter to expand_ref. About returning either child or modified context, I'll have to look into it.